object.cpp

```diff
@@ -756,7 +756,7 @@ std::string object_description(int obj)
     int buildable = 1;
     auto pS = FindSkill(o->skill);
     if (o->item == -1 || o->skill == nullptr || !pS || pS->get().flags & SkillType::DISABLED) buildable = 0;
-    if (o->item != I_WOOD_OR_STONE && (ItemDefs[o->item].flags & ItemType::DISABLED)) buildable = 0;
+    if (o->item >= 0 && o->item != I_WOOD_OR_STONE && (ItemDefs[o->item].flags & ItemType::DISABLED)) buildable = 0;
```

Comment:
I think the I_WOOD_OR_STONE check needs to come first since it's value is -2.  Or, it could be removed entirely as it's covered under the >= 0 check and thus not necessary here.

Reply:
I didn't realize that I_WOOD_OR_STONE was a special case. I dislike these negative constants that signify special cases in variables that are usually an index into an array, Too easy to mess up. Will fix, and clean up the surrounding code a bit.